test/python/algorithms/test_estimator_gradient.py

```diff
@@ -0,0 +1,330 @@
+    """Test Estimator Gradient"""
+
+    @combine(
+        grad=[FiniteDiffEstimatorGradient, ParamShiftEstimatorGradient, LinCombEstimatorGradient]
+    )
+    def test_gradient_p(self, grad):
+        """Test the estimator gradient for p"""
+        estimator = Estimator()
+        a = Parameter("a")
+        qc = QuantumCircuit(1)
+        qc.h(0)
+        qc.p(a, 0)
+        qc.h(0)
+        gradient = grad(estimator)
+        op = SparsePauliOp.from_list([("Z", 1)])
+        param_list = [[np.pi / 4], [0], [np.pi / 2]]
+        correct_results = [[-1 / np.sqrt(2)], [0], [-1]]
+        for i, param in enumerate(param_list):
+            values = gradient.run([qc], [op], [param]).result().values[0]
+            for j, value in enumerate(values):
+                self.assertAlmostEqual(value, correct_results[i][j], 3)
+
+    @combine(
+        grad=[FiniteDiffEstimatorGradient, ParamShiftEstimatorGradient, LinCombEstimatorGradient]
+    )
+    def test_gradient_u(self, grad):
+        """Test the estimator gradient for u"""
+        estimator = Estimator()
+        a = Parameter("a")
+        b = Parameter("b")
+        c = Parameter("c")
+        qc = QuantumCircuit(1)
+        qc.h(0)
+        qc.u(a, b, c, 0)
+        qc.h(0)
+        gradient = grad(estimator)
+        op = SparsePauliOp.from_list([("Z", 1)])
+
+        param_list = [[np.pi / 4, 0, 0], [np.pi / 4, np.pi / 4, np.pi / 4]]
+        correct_results = [[-0.70710678, 0.0, 0.0], [-0.35355339, -0.85355339, -0.85355339]]
+        for i, param in enumerate(param_list):
+            values = gradient.run([qc], [op], [param]).result().values[0]
+            for j, value in enumerate(values):
+                self.assertAlmostEqual(value, correct_results[i][j], 3)
+
+    @combine(
+        grad=[FiniteDiffEstimatorGradient, ParamShiftEstimatorGradient, LinCombEstimatorGradient]
+    )
+    def test_gradient_efficient_su2(self, grad):
+        """Test the estimator gradient for EfficientSU2"""
+        estimator = Estimator()
+        qc = EfficientSU2(2, reps=1)
+        op = SparsePauliOp.from_list([("ZI", 1)])
+        gradient = grad(estimator)
+        param_list = [
+            [np.pi / 4 for param in qc.parameters],
+            [np.pi / 2 for param in qc.parameters],
+        ]
+        correct_results = [
+            [
+                -0.35355339,
+                -0.70710678,
+                0,
+                0.35355339,
+                0,
+                -0.70710678,
+                0,
+                0,
+            ],
+            [0, 0, 0, 1, 0, 0, 0, 0],
+        ]
+        for i, param in enumerate(param_list):
+            values = gradient.run([qc], [op], [param]).result().values[0]
+            np.testing.assert_almost_equal(values, correct_results[i], 3)
+
+    @combine(
+        grad=[FiniteDiffEstimatorGradient, ParamShiftEstimatorGradient, LinCombEstimatorGradient]
+    )
+    def test_gradient_rxx(self, grad):
```

Comment:
Are the two-qubit Pauli rotation tests just random examples or do they test something special? These could also all be summarized in a single test with `@data` 🙂

Reply:
They are for linear combination gradients and the parameter shift gradients. The current implementation supports those 2 qubit gates (directly use these gates to calculate gradients instead of decomposing them)
https://github.com/a-matsuo/qiskit-terra/blob/9abbd2ac79ef1177308c236c0be344d2333384c6/qiskit/algorithms/gradients/utils.py#L285